pkg/kds/context/context.go

```diff
@@ -61,22 +61,47 @@ func DefaultContext(
 		config_manager.ClusterIdConfigKey: true,
 	}
 
-	mappers := []reconcile.ResourceMapper{
-		MapZoneTokenSigningKeyGlobalToPublicKey,
-		RemoveK8sSystemNamespaceSuffixFromPluginOriginatedResourcesMapper(
-			cfg.Store.Type,
-			cfg.Store.Kubernetes.SystemNamespace,
+	globalMappers := []reconcile.ResourceMapper{
+		UpdateResourceMeta(util.WithLabel(mesh_proto.ResourceOriginLabel, mesh_proto.ResourceOriginGlobal)),
+		If(
+			And(
+				TypeIs(system.GlobalSecretType),
+				NameHasPrefix(zone_tokens.SigningKeyPrefix),
+			),
+			MapZoneTokenSigningKeyGlobalToPublicKey),
+		If(
+			And(
+				IsKubernetes(cfg.Store.Type),
+				Not(TypeIs(system.ConfigType)), Not(TypeIs(system.SecretType)), Not(TypeIs(system.GlobalSecretType)),
+			),
+			RemoveK8sSystemNamespaceSuffixMapper(cfg.Store.Kubernetes.SystemNamespace)),
+		If(
+			And(
+				ScopeIs(core_model.ScopeMesh),
+				Not(TypeIs(system.SecretType)),
+			),
+			HashSuffixMapper(true)),
+	}
+
+	zoneMappers := []reconcile.ResourceMapper{
+		UpdateResourceMeta(
+			util.WithLabel(mesh_proto.ResourceOriginLabel, mesh_proto.ResourceOriginZone),
+			util.WithLabel(mesh_proto.ZoneTag, cfg.Multizone.Zone.Name),
```

Comment:
The problem with Delta XDS, when deleting resource we can receive only `ResourceKey`:
```go
type UpstreamResponse struct {
    RemovedResourcesKey []model.ResourceKey
}
```
so receiving side can't figure a hashed name because there are no labels.

Reply:
I [tried to move](https://github.com/kumahq/kuma/pull/8762/commits/1ae96de157a600e907c44b43c6dd2f0c645760a) mapping to global, but then [moved it back to zone].(https://github.com/kumahq/kuma/pull/8762/commits/f5e84730cae91b57400bc6984e39678a70193f67)

Reply:
I see. Can we have an additional validation to reject resources that tries to send us kuma.io/zone and origin with invalid data (missing/wrong value)?

Reply:
Talked offline that we can always call
```
				for _, r := range rs.GetItems() {
					r.SetMeta(util.CloneResourceMeta(r.GetMeta(),
						util.WithLabel(mesh_proto.ZoneTag, clusterName),
						util.WithLabel(mesh_proto.ResourceOriginLabel, mesh_proto.ResourceOriginZone),
					))
				}
```
OnResourceReceived